docs/zedctr.md

```diff
@@ -0,0 +1,96 @@
+
+## Location
+
+`zectr` is built in [pkg/zedctr](../pkg/zedctr) in this repository, although it includes certain outside packages when it is built. The resultant package is an OCI image named `zededa/zedctr`, which may be pushed to the image registry.
+
+## Runtime Config
+
+When run in the live rootfs, it runs with maximal capabilities, essentially as if it is running outside a container. These runtime capabilities are defined in [build.yml](../pkg/zedctr/build.yml).
+
+* It has `capabilities: all`, which should be split into individual capabilities listed for each service.
+* It has `pid: host`, working in the same PID namespace as the host.
+* It has `net: host`, working directly in the host networking namespace.
+* It mounts most of the root filesystem into its own namespace, including `/:/hostfs`. The mounts are listed below:
+
+```yml
+  binds:
+    - /lib/modules:/lib/modules
+    - /dev:/dev
+    - /etc/resolv.conf:/etc/resolv.conf
+    - /run:/run
+    - /var/config:/config
+    - /var/log:/var/log/dom0
+    - /:/hostfs
+```
+
+## Components
+
+The image `zededa/zedctr` is built from its [Dockerfile](../pkg/zedctr/Dockerfile), which is generated from its [Dockerfile.in](../pkg/zedctr/Dockerfile.in), with the relevant image names and tags patched in.
+
+It is built from the following packages, installed into the following locations:
+
+|Package|Image|Source|Install|Purpose|Comments|
+|---|---|---|---|---|
+|ztools|`zededa/ztools`|[go-provision](https://github.com/zededa/go-provision)|`/`|Interface with zedcloud||
+|lisp|`zededa/lisp`|[lisp](https://github.com/zededa/lisp)|[Locator/Identifier Separation Protocol](https://en.wikipedia.org/wiki/Locator/Identifier_Separation_Protocol)|see list [here](#lisp-install)|implementation for mesh networking|custom implementation in [go](https://golang.org), as the reference implementation is in python and has unneeded services|
+|xen tools|`zededa/xen-tools`|[pkg/xen-tools](../pkg/xen-tools)|`/`|Xen utilities||
+|dnsmasg|`zededa/dnsmasq`|[pkg/dnsmasq](../pkg/dnsmasq)|`/usr/sbin/dnsmasq` to `/opt/zededa/bin/dnsmasq`|local DNS server|Unclear why we need a local build|
+|strongswan|`zededa/strongswan`|[pkg/strongswan](../pkg/strongswan)|`/`|IPSec VPN|Dockerfile has [comment](../pkg/strongswan/Dockerfile#L63-L66), stating that we build custom because of libressl conflict with curl-dev. Should be investigated if this remains an issue|
+|gpt tools|`zededa/gpt-tools`|[pkg/gpt-tools](../pkg/gpt-tools)|`/`|Utilities for working with GPT partitioned filesystems. See [here](./UPSTREAMING.md#gpt-tools) for more details on what is in this package|
+|watchdog|`zededa/watchdog`|[pkg/watchdog](../pkg/watchdog)|`/usr/sbin` to `/usr/sbin`|[watchdog]() daemon to watch for userspace processes and, if they are not providing proper heartbeats, restart the system|Unclear why we need to restart the system vs restarting zombie processes, a la systemd/runit/supervisord/etc.|
```

Comment:
I can explain why we currently do a reboot. Voice?

Reply:
I will ping you on Slack.

Reply:
@rvs explained it to me. I have updated.